aten/src/ATen/native/cuda/Shape.cu

```diff
@@ -528,9 +529,8 @@ Tensor& cat_out_cuda(TensorList inputs, int64_t dimension, Tensor& out) {
   size[dimension] = cat_dim_size;
 
   // skip resizing if size of result is same as expected
-  if (out.sizes() != size) {
-    out.resize_(size, memory_format);
-  }
+  // raise a warning while resizing if output has one or more elements
+  at::native::resize_output(out, size, memory_format);
```

Comment:
Same caveat here as before

Reply:
Resolved in the conclusion to the discussion above.